src/providers/search.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import 'rxjs/add/operator/map';
+
+
+@Injectable()
+export class SearchProvider {
+
+  constructor(public http: Http) {
+    console.log('Hello Search Provider');
+  }
+
+
+  /**
+   * getSearchResults
+   */
+  public getSearchResults(query: string): Promise<Array<Object>> {
+    return new Promise<Array<Object>>((resolve, reject) => {
+      if (typeof query !== 'undefined') {
+        this.http.get(`https://v2.sg.media-imdb.com/suggests/${query[0]}/${query}.json`)
+          .subscribe(data => resolve(this._parseResponse(data['_body'], query)))
+      }
+      else {
+        resolve([]);
+      }
+    })
+  }
+
+  private _parseResponse(imdbResponse: string, query: string): Array<Object> {
+    let results = Array<Object>();
+    // replace spaces the query string with _
+    query = (query).split(' ').join('_')
```

Comment:
@aagavin, please, put a semicolon on each statement line in order to eliminate possible bugs. The same applies to lines 34, 36, 43 of this file